lib/Doctrine/Persistence/Mapping/Driver/DefaultFileLocator.php

```diff
@@ -136,7 +136,9 @@ public function getAllClassNames($globalBasename)
                     }
 
                     // NOTE: All files found here means classes are not transient!
-                    $classes[] = str_replace('.', '\\', $fileName);
+                    /** @psalm-var class-string */
```

Comment:
Using `assert(class_exists($class))` on this particular one makes the ORM test suite fail https://github.com/doctrine/orm/blob/2.8.x/tests/Doctrine/Tests/ORM/Tools/ConvertDoctrine1SchemaTest.php because there is heavy mocking involved.